lib/report_plugins/autotune.js

```diff
@@ -0,0 +1,333 @@
+  + '  width: 160px;'
+  + '  border-bottom: 1px solid black;'
+  + '  background: #b7b7b7'
+  + '}'
+  + '#autotune-report .autotune-report tr:nth-child(even) {'
+  + '  background: #efefef;'
+  + '}'
+  + '#autotune-report .autotune-report td {'
+  + '  width: 160px;'
+  + '  padding: 4px 0;'
+  + '}'
+
+autotune.prepareHtml = function autotunePrepareHtml(sorteddaystoshow) {
+};
+
+function appendMainRecords(table, header, current, recommended, translate) {
+  var tr = $('<tr/>');
+  if (current || recommended) {
+    $('<td>' + translate(header) + '</td>').appendTo(tr);
+    $('<td>' + current + '</td>').appendTo(tr);
+    $('<td>' + recommended + '</td>').appendTo(tr);
+  }
+  var td = $('<td/>')
+  appendArrowIcon(td, current, recommended);
+  td.appendTo(tr);
+  tr.appendTo(table);
+}
+
+function getCurrentISF(store) {
+  return store.sens[0].value;
+}
+
+function getCurrentCSF(store) {
+  return getCurrentISF(store) / getCurrentCarbRatio(store);
+}
+
+function getCurrentCarbRatio(store) {
+  return store.carbratio[0].value;
+}
+
+function roundNum(num) {
+  return Number(num).toFixed(3);
+}
+
+function appendArrowIcon(td, current, recommended) {
+  var icon = $('<img/>');
+  if (recommended > current) {
+    $('<img src="/images/up_arrow.png" />').appendTo(td);
+  } else if (recommended < current) {
+    $('<img src="/images/down_arrow.png" />').appendTo(td);
+  }
+}
+
+function findActiveProfiles(profiles, sorteddaystoshow) {
+  var result = {};
+  result.activeIndex = -1;
+  result.manyActive = false;
+
+  var dateFrom = new Date(sorteddaystoshow[0]);
+  var dateTo = new Date(sorteddaystoshow[sorteddaystoshow.length-1]);
+  var sortedProfiles = [];
+  for (var i = 0; i < profiles.length; i++) {
+    var record = {};
+    record.profileIndex = i;
+    if (! profiles[i].startDate) { continue; }
+    var splittedString = profiles[i].startDate.split("T");
+    record.startDate = new Date(splittedString[0]);
+    sortedProfiles.push(record);
+  }
+  sortedProfiles.sort(function (r1, r2) {
+    return (r1.startDate < r2.startDate) ? -1 : ((r1.startDate > r2.startDate) ? 1 : 0);
+  });
+  for (i = 0; i < sortedProfiles.length; i++) {
+    var profile = sortedProfiles[i];
+    if (profile.startDate <= dateFrom) {
+      result.activeIndex = profile.profileIndex;
+      continue;
+    }
+    if (profile.startDate >= dateFrom && profile.startDate <= dateTo)
+      result.manyActive = true;
+    if (profile.startDate > dateFrom)
+      break;
+  }
+
+  return result;
+}
+
+function getHourMinuteText(hour, minute)
+{
+  var text = "";
+  if (hour < 10)
+    text += "0";
+  text += hour.toString();
+  text += ":";
+  if (minute < 10)
+    text += "0";
+  text += minute.toString();
+  return text;
+}
+
+function findBasalRecord(data, time)
+{
+  for(i = 0; i < data.length; i++)
+  {
+    if (data[i].time == time)
+      return data[i];
+  }
+  return null;
+}
+
+function findRecommendedRecord(data, minutes) {
+  for(i = 0; i < data.length; i++)
```

Comment:
same here
`for(var i = 0; i < data.length; i++)`

Reply:
Fixed both in e204200.  Thanks.